### ISSUE TITLE
Website: Pictures Organisation

### ISSUE THREAD

**Issue by fczaja** [odoo/odoo#12533]
## Impacted versions
- 9.0
## Goal

Images should be better organized. For this purpose it must be possible to create folders in which the images can be saved. The image name is also the alt text. 
## Reason

Working with images should be facilitated. These include that they can be better organized to find images fast.
## Current behavior

Currently, all images are stored in a folder.
## Expected behavior

The creation of new pictures folders should be possible.
The creation of images folders in the Pictures folder should be possible.
Pictures folder in which there are images, can not be deleted (for safety).
The image name is also the alt text. 
The alt text can be changed anyway.
## This information is available

https://github.com/odoo/odoo/issues/12527

**Comment by pedrobaeza** [comment]
This has nothing to do with SEO, but with facilities for the website publisher, so please don't put an incorrect title.

I don't agree that's the way to go. Current approach in v11 includes a search bar for looking for a resource through its name, so that's going to avoid a lot of duplicates through several folders.

I miss though 3 features:

* Possibility of renaming the images text.
* When uploading a image, automatically compare if there's already the same image previously uploaded (through MD5 or similar method), and reuse it.
* Put a warning before removing an image if it's used anywhere.

@JKE-be do you think any of these features can be feasible?

**Comment by JKE-be** [comment]
Some improvements has been done:

You have now the searchbar,
You only see filtered images and not all

You can also change the "title", "alt" attribute on an image into the editor. (And the image path will become translatable in the v12 - not yet merged but we have the patch)
The filename could be an improvement since it is only SEO and that controller already allow it. Basically you can already now add the image now in the image path.

About the upload, that will just make the UI cleaner, but anyway since the hash of the file will be the same, that will be not stored twice.

The point about the image deleted tracker, is technically hard to do. All what we can do it a similary logic that when you delete a page. Search in ir_ui_view/blog/forum if image seems to be used. But we cannot warranty it, since there are several route / way to display a same picture.

**Comment by pedrobaeza** [comment]
Thanks for the follow-up
